app/src/main/java/edu/grinnell/grinnell_publications_android/Controllers/MainRecyclerViewAdapter.java

```diff
@@ -1,24 +1,54 @@
 package edu.grinnell.grinnell_publications_android.Controllers;
 
+import android.app.Activity;
 import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
 import android.view.ViewGroup;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
- * Created by prabir on 4/24/16, AppDev Grinnell.
+ *
  */
-public class MainRecyclerViewAdapter extends RecyclerView.Adapter {
+public class MainRecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewHolder> {
+
+    private List<RecyclerItem> data;
+    private Activity activity;
+
+    public MainRecyclerViewAdapter(Activity activity) {
+        this.data = new ArrayList<>();
+    }
+
+    /**
+     * Update the data for the recycler view
+     *
+     * @param data a list of recycler items that will populate the recycler view
+     */
+    public void setData(List<RecyclerItem> data) {
+        this.data = data;
+        notifyDataSetChanged();
+    }
+
+
     @Override
-    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+    public RecyclerViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        LayoutInflater li = activity.getLayoutInflater();
```

Comment:
Rename the LayoutInflater instance to something more appropriate eg. inflater, layoutInflater